src/main/java/com/example/demo/service/CartServiceImpl.java

```diff
@@ -0,0 +1,101 @@
+package com.example.demo.service;
+
+import com.example.demo.dto.cart.CartDto;
+import com.example.demo.dto.item.CreateItemRequestDro;
+import com.example.demo.dto.item.UpdateItemRequestDto;
+import com.example.demo.exception.EntityNotFoundException;
+import com.example.demo.mapper.CartMapper;
+import com.example.demo.model.Book;
+import com.example.demo.model.CartItem;
+import com.example.demo.model.ShoppingCart;
+import com.example.demo.model.User;
+import com.example.demo.repository.BookRepository;
+import com.example.demo.repository.CartItemRepository;
+import com.example.demo.repository.ShoppingCartRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class CartServiceImpl implements CartService {
+    private final ShoppingCartRepository cartRepository;
+    private final CartMapper cartMapper;
+    private final BookRepository bookRepository;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    public CartDto getCart(Authentication authentication) {
+        User user = (User) authentication.getPrincipal();
+        return cartMapper.toDto(cartRepository.getCart(user));
+    }
+
+    @Transactional
+    @Override
+    public CartDto addItemToCart(CreateItemRequestDro requestDto,
+                                 Authentication authentication) {
+        User user = (User) authentication.getPrincipal();
+        ShoppingCart shoppingCart = cartRepository.getCart(user);
+
+        Book book = bookRepository.findById(requestDto.getBookId())
+                .orElseThrow(() -> new EntityNotFoundException(
+                        "Book not found with ID: " + requestDto.getBookId()));
+
+        CartItem cartItem = shoppingCart.getCartItems().stream()
+                .filter(item -> item.getBook().getId().equals(book.getId()))
+                .findFirst()
+                .orElse(null);
```

Comment:
In method `cartRepository.getCart(user)` you also need join fetch cart items, because you access it here and it cause lazy loading